src/genius/services/http-provider/http-data-provider.service.ts

```diff
@@ -0,0 +1,178 @@
+
+  public persons: Array<Person>;
+
+  constructor(private http: HttpClient) {
+    super();
+    this.getPersons().subscribe(httpResponseGetPersons => this.persons = httpResponseGetPersons,
+      (errorHttpResponseGetPersons) => {
+        console.error(`Error status: ${errorHttpResponseGetPersons.error?.status}\n Error message: ${errorHttpResponseGetPersons.error?.message}\n Error path: ${errorHttpResponseGetPersons.error?.path}\n`);
+      });
+  }
+
+  protected mapPerson(obj: any): Person {
+    let person = new Person();
+    person.id = obj.id;
+    person.firstName = (obj.name) ? obj.name.first : null;
+    person.middleName = (obj.name) ? obj.name.middle : null;
+    person.lastName = (obj.name) ? obj.name.last : null;
+    person.sex = (obj.gender) ? obj.gender.toLowerCase() : null;
+    person.familyId = obj.parentFamilyId;
+
+    return person;
+  }
+
+  protected mapFamily(obj: any): Family {
+    let family = new Family();
+    family.children = [];
+    family.id = obj.id;
+    family.mother = this.persons.find(person => obj.wife === person.id)
+    family.father = this.persons.find(person => obj.husband === person.id)
+    if (obj.children && obj.children.length > 0) {
+      obj.children.forEach((childId: number) => {
+        let target = this.persons.find(person => person.id === childId)
+        if (target) family.children.push(target)
+      })
+    }
+    return family;
+  }
+
+  public addNewFamily(family: Family): Observable<Object> {
+    const httpOptions = {
+      headers: new HttpHeaders({
+        'Content-Type': 'application/json',
+        'Accept': 'application/json'
+      })
+    };
+
+    let newFamily: FamilyDTO = new FamilyDTO();
+    if (family.father) {
+      newFamily.husband = family.father.id
+    }
+    if (family.mother) {
+      newFamily.wife = family.mother.id
+    }
+    if (family.children) {
+      newFamily.children = family.children.map(child => child.id);
+    }
+
+    if (family.children && family.children.length > 0) {
+      family.children.forEach(child => newFamily.children.push(child.id))
+    }
+
+    return this.http.post(`${environment.url}/families`, newFamily, httpOptions)
+  }
+
+  public addNewPerson(person: Person): Observable<Object> {
+    const httpOptions = {
+      headers: new HttpHeaders({
+        'Content-Type': 'application/json',
+        'Accept': 'application/json'
+      })
+    };
+    let newPerson = new PersonDTO();
+    newPerson.name = {};
+    if (person.firstName) {
+      newPerson.name.first = person.firstName;
+    }
+    if (person.lastName) {
+      newPerson.name.last = person.lastName;
+    }
+    if (person.middleName) {
+      newPerson.name.middle = person.middleName;
+    }
+    if (person.sex) {
+      newPerson.gender = person.sex.toUpperCase();
+    }
+
+    return this.http.post(`${environment.url}/persons`, newPerson, httpOptions)
+  }
+
+  public changeFamily(family: Family): Observable<Object> {
+    let changeFamily = new FamilyDTO()
```

Comment:
missing content type of data you are sending to server. By default httpClient use `json`, but it's recomended to specify it explicitly